pkg/factorymodule/factory.go

```diff
@@ -0,0 +1,130 @@
+package factorymodule
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/mir/pkg/events"
+	"github.com/filecoin-project/mir/pkg/logging"
+	"github.com/filecoin-project/mir/pkg/modules"
+	"github.com/filecoin-project/mir/pkg/pb/eventpb"
+	"github.com/filecoin-project/mir/pkg/pb/factorymodulepb"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+// TODO: Add support for active modules as well.
+
+type moduleGenerator func(id t.ModuleID, params *factorymodulepb.GeneratorParams) modules.PassiveModule
+
+type FactoryModule struct {
+	ownID     t.ModuleID
+	generator moduleGenerator
+
+	submodules      map[t.ModuleID]modules.PassiveModule
+	moduleRetention map[t.RetentionIndex][]t.ModuleID
+	retIdx          t.RetentionIndex
+
+	logger logging.Logger
+}
+
+func NewFactoryModule(id t.ModuleID, generator moduleGenerator, logger logging.Logger) *FactoryModule {
+	return &FactoryModule{
+		ownID:     id,
+		generator: generator,
+
+		submodules:      make(map[t.ModuleID]modules.PassiveModule),
+		moduleRetention: make(map[t.RetentionIndex][]t.ModuleID),
+		retIdx:          0,
+
+		logger: logger,
+	}
+}
+
+func (fm *FactoryModule) ImplementsModule() {
+}
```

Comment:
I think such functions are usually written in one line.
```go
func (fm *FactoryModule) ImplementsModule() {}
```

Reply:
Fixed.